cortex/primary/significant_locations.py

```diff
@@ -57,6 +57,34 @@ def _euclid(lat, lng, lat0, lng0):  # degrees -> km
         return 110.25 * ((((lat - lat0) ** 2) + (((lng - lng0) * np.cos(lat0)) ** 2)) ** 0.5)
     return _euclid(g0[0], g0[1], g1[0], g1[1])
 
+
+def remove_clusters(clusters, MAX_DIST=300):
+    """ Function to remove clusters that are less than specified distance (MAX_DIST) away from at least one other cluster
+
+        Args:
+            clusters: (list of dicts) Each dict in the list is a significant location 
+            MAX_DIST: (int) Maximum distance allowed between clusters (in meters)
+    """
+    n = len(clusters)
+    clusters_removed = []
+    for i in range(n - 1):
+        big_cluster_prop = clusters[i]['proportion']
+        if big_cluster_prop > 0:
+            big_cluster_latlon = (clusters[i]['latitude'], clusters[i]['longitude'])
+            for j in range(i + 1, n):
+                small_cluster_latlon = (clusters[j]['latitude'], clusters[j]['longitude'])
+                dist = get_hometime.distance(big_cluster_latlon, small_cluster_latlon)
+                if dist < MAX_DIST:
+                    clusters[i]['proportion'] += clusters[j]['proportion']
+                    clusters[j]['proportion'] = 0
+                    clusters_removed.append(j)
```

Comment:
Shouldn't the duration in each cluster also be updated here?

Reply:
Yeah sorry I removed duration while testing, I just added the duration in the recent commit. Removed the clusters_removed since it is not being returned or used in function